File: src/games/LegendaryGame.js
const { Game } = require("./Game.js");

/**
 * A class representing a legendary games launcher game
 * @property {string} appName - The game's epic games launcher app name
 * @property {LegendaryGameProcessContainer} processContainer - The game's process container
 */
class LegendaryGame extends Game {

	platform = "PC";
	isInstalled = true; // Legendary only exposes installed games

	/**
	 * Create a legendary games launcher game
	 * @param {string} appName - The game's app name
	 * @param {string} name  - The game's displayed name
	 */
	constructor(appName, name) {
		super(name);
		this.appName = appName;
	}

	/**
	 * Create a string representation of the game
	 * @returns {string} - A string representing the game
	 */
	toString() {
		return `${this.name} - ${this.source} - ${this.appName}`;
	}

}

module.exports = {
	LegendaryGame
};

File: gali/games/legendary_game.py
from dataclasses import dataclass, field

from gali.games.game import Game

@dataclass
class LegendaryGame(Game):

	platform : str = field(default="PC", init=False)
	app_name : str = field(default=None)

	def get_start_command(self, **kwargs) -> tuple[str]:
		args = ["legendary", "launch", self.app_name]
		return tuple(args)